### PR TITLE
Adding analyzer, connector service objects with their unit tests

### PR DESCRIPTION
**Pull request by fear-the-reaper**
# Description
    Added the analyzer, and connector service objects with their respective unit tests

## Issues
#16 

## Type of change
list down the type of change
- [ ] Bug fix
- [x] New feature (added a new endpoint or created)
- [ ] Breaking fix (a bug fix or new feature that would cause exisiting functionality not to work as expected)

## Checklist
- [ ] if a new feature was added it passed all its (if made) test cases 
- [ ] if a bug fix was added it passed all its existing test cases 

### Important Rules
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem
- Every time you make changes to the PR and you think the work is done, you should explicitly ask for a review

Please delete if the PR is for bug fixing.

### CODE REVIEW

File: gointelowl/analyzer.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	return status.Status, nil
+}
+
+func (analyzerConfig *AnalyzerConfig) Display() error {
+	jsonBytes, err := json.Marshal(analyzerConfig)
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(jsonBytes))
+	return nil
```
Comment (sp35) [inline gointelowl/analyzer.go]: Is this method needed? I pointed this out in tag service pr as well!

File: gointelowl/connector.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+func (connectorConfig *ConnectorConfig) Display() error {
+	jsonBytes, err := json.Marshal(connectorConfig)
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(jsonBytes))
+	return nil
+}
```
Comment (sp35) [inline gointelowl/connector.go]: Re: Is this method needed?

File: tests/connectorService_test.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
...ass your request through a proxy server."},"tlp":{"value":{"type":"white","color":"#ffffff","x_opencti_order":1},"type":"dict","description":"Change this as per your organization's threat sharing conventions."}},"verification":{"configured":false,"error_message":"(api_key_name,url_key_name) not set; (0 of 2 satisfied)","missing_secrets":["api_key_name","url_key_name"]},"maximum_tlp":"WHITE"},"YETI":{"name":"YETI","python_module":"yeti.YETI","disabled":true,"description":"find or create observable on YETI, linking the successful analysis on IntelOwl.","config":{"queue":"default","soft_time_limit":30},"secrets":{"api_key_name":{"env_var_key":"CONNECTOR_YETI_KEY","description":"API key for your YETI instance","required":true},"url_key_name":{"env_var_key":"CONNECTOR_YETI_URL","description":"API URL of your YETI instance","required":true}},"params":{"verify_ssl":{"value":true,"type":"bool","description":"Enable SSL certificate server verification. Change this if your YETI instance has not SSL enabled."}},"verification":{"configured":false,"error_message":"(api_key_name,url_key_name) not set; (0 of 2 satisfied)","missing_secrets":["api_key_name","url_key_name"]},"maximum_tlp":"WHITE"}}`
```
Comment (sp35) [inline tests/connectorService_test.go:16]: Better handling of mocks for service tests

File: tests/connectorService_test.go [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+				diff := cmp.Diff(testCase.Want, (*gottenConnectorConfigList))
+				if diff != "" {
+					t.Fatalf(diff)
+				}
+			}
+		})
+	}
+}
+
+// func TestConnectorServiceHealthCheck(t *testing.T) {
```
Comment (sp35) [inline tests/connectorService_test.go]: Are you adding this test to this PR?

**Review by sp35 — COMMENTED** [review]
Overall LGTM!

File: tests/connectorService_test.go (hunk 3 above)
Comment (sp35) [reply tests/connectorService_test.go:16]: Creating an issue for this!

File: gointelowl/analyzer.go (hunk 1 above)
Comment (fear-the-reaper) [reply gointelowl/analyzer.go]: Well in my opinion a custom Display method should be applied so we can easily pretty print the data. @sp35 and @eshaan7 what do you guys think?

File: gointelowl/connector.go (hunk 2 above)
Comment (fear-the-reaper) [reply gointelowl/connector.go]: Re: a custom Display method should be applied so we can easily pretty print the data.

File: tests/connectorService_test.go (hunk 4 above)
Comment (fear-the-reaper) [reply tests/connectorService_test.go]: Yes! I'll uncomment this out my bad.

File: gointelowl/analyzer.go (hunk 1 above)
Comment (fear-the-reaper) [reply gointelowl/analyzer.go]: It is not a requirement as we are making a Library not a CLI!

**Review by sp35 — APPROVED** [review]

**Pull request merged**